src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3249 @@
+                        keyword.HasReservedProperties = true;
+                        continue;
+                    }
+
+                    // Otherwise, add it to the Keyword List.
+                    var keyProp = new System.Management.Automation.Language.DynamicKeywordProperty();
+                    keyProp.Name = prop.Name;
+
+                    // Copy the type name string. If it's an embedded instance, need to grab it from the ReferenceClassName
+                    bool referenceClassNameIsNullOrEmpty = string.IsNullOrEmpty(prop.ReferenceClassName);
+                    if (prop.CimType == "Instance" && !referenceClassNameIsNullOrEmpty)
+                    {
+                        keyProp.TypeConstraint = prop.ReferenceClassName;
+                    }
+                    else if (prop.CimType == "InstanceArray" && !referenceClassNameIsNullOrEmpty)
+                    {
+                        keyProp.TypeConstraint = prop.ReferenceClassName + "[]";
+                    }
+                    else
+                    {
+                        keyProp.TypeConstraint = prop.CimType.ToString();
+                    }
+
+                    // Check to see if there is a Values attribute and save the list of allowed values if so.
+                    var values = prop.Qualifiers?.Values;
+                    if (values != null)
+                    {
+                        foreach(var val in values)
+                        {
+                            keyProp.Values.Add(val.ToString());
+                        }
+                    }
+
+                    // Check to see if there is a ValueMap attribute and save the list of allowed values if so.
+                    var nativeValueMap = prop.Qualifiers?.ValueMap;
+                    List<string> valueMap = null;
+                    if (nativeValueMap != null)
+                    {
+                        valueMap = new List<string>();
+                        foreach(var val in nativeValueMap)
+                        {
+                            valueMap.Add(val.ToString());
+                        }
+                    }
+
+                    // Check to see if this property has the Required qualifier associated with it.
+                    if (string.Equals(prop.Qualifiers?.Required?.ToString(), "True", StringComparison.OrdinalIgnoreCase))
+                    {
+                        keyProp.Mandatory = true;
+                    }
+
+                    // Check to see if this property has the Key qualifier associated with it.
+                    if (string.Equals(prop.Qualifiers?.Key?.ToString(), "True", StringComparison.OrdinalIgnoreCase))
+                    {
+                        keyProp.Mandatory = true;
+                        keyProp.IsKey = true;
+                    }
+
+                    // set the property to mandatory is specified for the resource.
+                    if (runAsBehavior == DSCResourceRunAsCredential.Mandatory)
```

Comment:
We don't support runAs with PS7?

Reply:
`PSDesiredStateConfiguration` module currently throws 'PsDscRunAsCredentialNotSupport,Invoke-DscResource'